text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,370 @@
+This approach is considered decentralized as it does not impose any direct
+collaboration between parties while ensuring consistency between versions of
+dependencies and their associated telemetry schemas.
+
+#### Centralized Approach
+
+In this approach, the library designers have no interaction with the external
+world. They define a telemetry schema for their library, version the resolved
+schema in their repository, and release their library as usual.
+
+On the application and service owners' side, they must either make a local copy
+of the resolved telemetry schemas corresponding to their dependencies (similar
+conceptually to the model followed by protobuf and gRPC) or link to the resolved
+telemetry schema for a specific version. In both cases, the synchronization
+between the versions of the library and the resolved telemetry schema is the
+responsibility of the application owner.
+
+![Centralized Approach](./img/0240-centralized-approach.png)
+
+This approach is considered centralized as the application owner has the
+responsibility of maintaining, in a centralized manner locally, all the
+telemetry schemas of the dependencies.
+
+A specific OTEP dedicated to the resolved telemetry schema will be responsible
+for defining the format and structure of the resolved telemetry schema, as well
+as the best approach to collect the resolved telemetry schemas of the different
+dependencies.
+
+## Roadmap
+
+To facilitate the review process and progressively deliver value to the
+OpenTelemetry community, a series of OTEPs and tools are proposed.
+
+### OTEPs
+
+* Telemetry Schema versions (incremental steps):
+  * V1.1.1: Add a section to import one or several Semantic Convention
+  Registries. Once implemented, the official Semantic Convention Registry will
+  be referenced from the official OpenTelemetry Root Telemetry Schema.
+  * V1.2: Add references to parent schemas and define a schema section to
+  describe application/library signals. In this version, only metrics, logs, and
+  spans signals are described. This is the first version of the telemetry schema
+  that can be used by the Client SDK generator. In this version only one schema
+  can be inherited. More complex hierarchies will be supported in version 1.5.
+  * V1.3: Add support for events in the schema section. This OTEP is contingent
+  upon the approval of events in OpenTelemetry.
+  * V1.4: Add support for metric_groups (multivariate metrics) in the schema
+  section. This will only be relevant if a support for multivariate metrics is
+  planned in the Client SDK Generator.
+  * V1.5: Add support for multi-inheritance and for schema hierarchies of more
+  than 2 levels. This is useful for supporting complex environments and will
+  allow the distribution of semantic convention registries across vendors and
+  enterprises. This OTEP depends on the “Schema Child/Parent relationship
+  resolution” OTEP.
+* Resolved Schema Format v1.0. This OTEP will define both the structure and
```

Comment:
I would like to consider flipping this the other way round. 

Instead of introducing a new concept of "resolved schema" I want to say that what we have today _is_ the resolved schema. That's what we publish today and what we will continue publishing in the future (with more features in newer versions). The "unresolved" schema is the new concept that we will introduce which possibly may even exist only internally at Otel and not intended for public consumption (unless we see good reasons for such consumption). The unresolved schema + semantic conventions will be the input to an internal tool that produces the Schema Files as we know them (and which are always resolved).

Reply:
Yes, I understand the rationale and it doesn't seem to be a significant modification. It will require me to make some minor adjustments here and there in the OTEP. I will take care of that as soon as possible.

Reply:
I fully agree with @tigrannajaryan, this also takes care of https://github.com/open-telemetry/oteps/pull/243/files#r1413837539.

Publishing resolved schemas would allow us to go without requiring a public specification of the attribute registry, which would be required if users were required to resolve schemas. The schema being the single point where users interface with semantic conventions makes life easier for users, as well as for spec authors.